cmd/finch/nerdctl_remote.go

```diff
@@ -501,3 +539,134 @@ func handleEnvFile(fs afero.Fs, systemDeps NerdctlCommandSystemDeps, arg, arg2 s
 	}
 	return skip, envs, nil
 }
+
+func handleCache(fc *config.Finch, arg string) string {
+	// Hack to handle consistency params during mounts. This is assuming no other commands or env variable will have the word consistency.
+	if fc == nil || fc.Mode == nil || *fc.Mode != "dockercompat" || runtime.GOOS != "darwin" {
+		logrus.Warn("handleCache returning unchanged: ", arg)
+		logrus.Warn("handleCache info: ", fc.Mode, *fc.Mode, runtime.GOOS)
+		return arg
+	}
+
+	if strings.Contains(arg, "consistency") {
+		logrus.Warn("handleCache manipulating: ", arg)
+		arg = strings.Replace(arg, ",consistency=cached", "", 1)
+		arg = strings.Replace(arg, ",consistency=delegated", "", 1)
+		arg = strings.Replace(arg, ",consistency=consistent", "", 1)
+		logrus.Warn("handleCache returning: ", arg)
+	}
+	return arg
+}
+
+func handleLoad(fc *config.Finch, args []string) []string {
+	if fc == nil || fc.Mode == nil || *fc.Mode != "dockercompat" || args == nil {
+		return args
+	}
+
+	if *fc.Mode == "dockercompat" {
+		logrus.Warn("appending --output-type!!")
+		logrus.Warn("args before appending", args)
+		args = append(args, "--output=type=docker")
+		logrus.Warn("args after appending", args)
+	}
+
+	return args
+}
+
+func handleDockerCompatInspect(fc *config.Finch, args []string) []string {
+	if fc == nil || fc.Mode == nil || *fc.Mode != "dockercompat" {
+		return args
+	}
+
+	newArgList := []string{}
+	isInspect := false
+	inspectIndex := -1
+	skipNext := false
+	inspectType := ""
+
+	for idx, arg := range args {
+		if skipNext {
+			skipNext = false
+			continue
+		}
+		if arg == "inspect" {
+			isInspect = true
+			inspectIndex = idx
+			newArgList = append(newArgList, arg)
+		}
+
+		if strings.Contains(arg, "--type") && (idx < len(args)-1) && isInspect {
+			if strings.Contains(arg, "=") {
+				inspectType = strings.Split(arg, "=")[1]
+			} else {
+				inspectType = args[idx+1]
+			}
+			if arg == "--type" {
+				skipNext = true
+			}
+			continue
+		}
+		newArgList = append(newArgList, arg)
+	}
+
+	if !isInspect {
+		return args
+	}
+
+	switch inspectType {
+	case "container":
+		dcTrailingArgs := append([]string{"--mode=dockercompat"}, newArgList[inspectIndex+2:]...)
+		dcMidArgs := append([]string{inspectType}, append([]string{newArgList[inspectIndex+1]}, dcTrailingArgs...)...)
+		newArgList = append(newArgList[:inspectIndex], dcMidArgs...)
+	case "":
+		break
+	default:
+		newArgList = append(newArgList[:inspectIndex], append([]string{inspectType}, newArgList[inspectIndex+1:]...)...)
+	}
+
+	return newArgList
+}
+
+func handleBuildx(fc *config.Finch, limaArgs []string) (bool, []string) {
```

Comment:
am thinking if we should first check for buildx and create a flag so that we dont enter this loop again, we already have all the info from the first loop

Reply:
the intial arg loop

Reply:
Agreed, the recent update has refactored into a handleBuildx mechanism and modifies-in-place the "buildx <sub_cmd>" scenarios.